### ISSUE TITLE
New Session About Hans-on F#

### ISSUE THREAD

**Issue by kirisky** [amplifying-fsharp/sessions#53]
### Session on Hans-on F#
Dong, from Individual Contributor. In EET/EEST, would like to tackle:
Onboarding to F# world and hands-on experience

#### Extra
This will be a project-based demonstration aimed at showcasing the F# ecosystem and best practices by building a real-world application. 

This project-based demonstration will showcase the F# ecosystem and best practices by building a real-world application. The goal is to help newcomers become familiar with the F# ecosystem and best practices.

##### Dependency

This will happen after [Session About Solving Good First Issues for FSharp](https://github.com/amplifying-fsharp/sessions/issues/52)

**Comment by nojaf** [comment]
Thanks for creating this! I think there are two key aspects to consider:
- How would you create a larger scale application? This includes organizing code, setting up tests, linting, documentation generation, and more.
- How do I navigate the .NET and F# ecosystem? This could be interesting to explore, as I often find it to be a source of frustration.

Which aspect would you be most eager to explore?

**Comment by kirisky** [comment]
Hi @nojaf! Thank you so much for the feedback! It is really helpful for refining the session plan.

I'd like to cover both, but doing it all in one session might make it too long to follow. So, I am thinking of breaking it up into several shorter sessions. 

----

> How would you create a larger scale application? This includes organizing code, setting up tests, linting, documentation generation, and more.

I completely agree with you. The project will be small to small-medium in scale. Ideally, I’d like to cover all the topics you mentioned as part of this project.

My current plan is to create something straightforward, like a simple web crawler or a CMS/CRM. These types of projects are easier to keep within a manageable scope, widely understood in the real world, and provide a great way to showcase F#'s ecosystem. That'll be able to demonstrate commonly used libraries and components, such as API services, databases, caching JSON handling, parallel processing, and data display.

I am planing to structure the sessions as followings:

- **1st session**
  - Set up the project and integrate the necessary libraries. By the end, we will have a working example, like crawling Wikipedia pages and displaying the data   
- **2nd session**
  - Refactor the project, applying best practices for code quality and maintainability   
- **3rd session**
  - Show how to deploy the project to Azure using F#-related tools (FARMER, AzureCLI, or others)

If you have any suggestions on topics or how to organize the sessions, **please feel free to share your thoughts as it would be extremely helpful in planning the sessions**.  I would greatly appreciate any ideas or suggestions from you or the community to help shape these sessions.


> How do I navigate the .NET and F# ecosystem? This could be interesting to explore, as I often find it to be a source of frustration.

Could you provide more details on this, Flo? What scenarios cause frustration in your case?

**Comment by nojaf** [comment]
Thanks for the enthusiasm, @kirisky! My initial reaction is that three sessions might be quite demanding for you to start. You may want to focus on the first session without overpromising the audience that this is a trilogy. It might be helpful to outline what you aim to cover in the first session so we can discuss the various aspects in more detail.

Another suggestion is to reach out to the maintainers of the different tools you'll be using to learn how they recommend utilizing them in 2024. For instance, if you use Fable, consider messaging Maxim to inquire about the recommended or active packages in Fable these days.

It could also be beneficial to start from a template (SAFE, SAFEr, mini scaffold) and adjust as needed.

> Could you provide more details on this, Florian? What scenarios cause frustration in your case?

I'm currently working with OCaml. Learning the language isn't really the hard part for someone coming from F#, but I find it challenging to grasp the build system, package management, switches, and which unit test framework to use. This is similar for anyone transitioning from another ecosystem to F#. The .NET CLI, fsproj files, NuGet—all these elements have a learning curve.

**Comment by kirisky** [comment]
Thank you so much, @nojaf! I'll adjust the content capacity based on the information and suggestions you shared.

**Comment by dawedawe** [comment]
Hey @kirisky , are you still interested in having a session? Maybe Nov 22nd or early December?

**Comment by kirisky** [comment]
> Hey @kirisky , are you still interested in having a session? Maybe Nov 22nd or early December?

Hi Dawe! Yeah, I am still.  But I have not been ready. If it is needed, I can prioritize this one as my first session. 

I would prefer to do the session in the early of December as this month is not available for me to prepare this.

What do you say?😃

**Comment by dawedawe** [comment]
Sounds great. Thanks!
How about December 6th?

**Comment by kirisky** [comment]
Yeah, it works for me. 

If there’s anything I should be aware of for this session, please let me know in advance!

Thank you, @dawedawe! 😄

**Comment by dawedawe** [comment]
Awesome, we'll prepare a session page PR for you to review containing some description of the session. Similar to the ones here: https://amplifyingfsharp.io/sessions/
Then we should have a short test zoom call in the days before the session to make sure the streaming setup works from your machine.

**Comment by kirisky** [comment]
Thanks, Dawe! 

Will take a look other sessions as well for preparing mine. If you have any other recommended resource, just let me know. :)

**Comment by nojaf** [comment]
Hello @kirisky, how are things here?
Do you think you can give a session this month?

**Comment by kirisky** [comment]
Hi @nojaf! Yeah, I can do it. However, I’m quite busy at the moment, so scheduling my session for next week would work better for me. That said, if no one else is available, I’m happy to do the presentation this Friday. Let me know what works best!

**Comment by nojaf** [comment]
Well, there is no hard requirement that there is a session this week.
Taking https://github.com/amplifying-fsharp/sessions/issues/57 into account, would Friday December 20th work for you?

**Comment by kirisky** [comment]
Yes, that would be better! Thank you, and apologize for the postponement.

**Comment by nojaf** [comment]
Great! Whenever you are ready, please send a PR to https://github.com/amplifying-fsharp/amplifying-fsharp.github.io to add your session page.

**Comment by kirisky** [comment]
Will do it whenever ready! Thanks for the link! 👍

**Comment by nojaf** [comment]
Hello @kirisky, following up on this, will you be able to give a session this Friday?

**Comment by kirisky** [comment]
Hi @nojaf! Yeah, I can do it this Friday. Will prepare for it. :)

It will be an introduction to the web development stack in the F# ecosystem, it covers essential components such as build systems, package management, testing frameworks, and more. The goal is to shorten the learning curve and empower beginners to start their projects efficiently.

And the title could be 'Hands-On F# Programming: First Web Project in F#'.

**Comment by nojaf** [comment]
Alright, that sounds good. 
I'm going to ask you to create a page over at https://github.com/amplifying-fsharp/amplifying-fsharp.github.io so we have something we can share on the socials.

What time did we settle on?

**Comment by kirisky** [comment]
Hi @nojaf! Thanks! Here is the PR: https://github.com/amplifying-fsharp/amplifying-fsharp.github.io/pull/275

This is my first time, I am unsure what time is suitable. The session is usually hosted at 14:00 GMT, right? 

And do we need any rehearsal?

**Comment by nojaf** [comment]
If the 14:00 GMT slot works for you that would be ideal.
We don't need to rehearse the entire session, but I would like to jump on a call with you today (Wednesday) or tomorrow if that works.

**Comment by kirisky** [comment]
Hi @nojaf! Unfortunately, I won’t be able to give the session this Friday due to an unexpected assignment from my company. 😢 It seems I’ll need to postpone it to next year. Apologize for that! 🙏

Let’s find a new time! I’m looking forward to having a call with you once we finalize the session content in the [PR](https://github.com/amplifying-fsharp/amplifying-fsharp.github.io/pull/275#issuecomment-2550600558).

**Comment by nojaf** [comment]
Alright, let's circle back to this next year! Good luck with your assignment!
